### PR TITLE
chore: adding queue to backbone

### PR DESCRIPTION
**Pull request by mrasif**
## Relevant Documentation Links
-

## Relevant Jira Links
-
-

## Other Relevant PRs
-

## Changes

(Description of changes)

### Changes Include

-   [ ] Bug fix 🐛 (non-breaking change which fixes an issue)
-   [x] New feature ✨ (non-breaking change which adds functionality)
-   [ ] Breaking change 😢 (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Refactor ⚒️ (code change that neither fixes a bug nor adds a feature)
-   [ ] Tooling ⚙️ (Changes to the build process or auxiliary tools and libraries such as documentation generation)
-   [ ] Performance ⚡ (code change that improves performance)

### Checklist

-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] My changes generate no new warnings
-   [ ] Requires migrations
-   [ ] Requires dependency update
-   [ ] Breaking API Change
-   [ ] I have added proper logs for production debugging
-   [ ] I have added performance (APM) monitoring checkpoints
-   [ ] I have added unit tests

## Screenshots (if any)
(prefer animated gif)

### Test #1

Description

- [ ] Acceptance Creteria #1
- [ ] Acceptance Creteria #2

### CODE REVIEW

File: utils/event.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package utils
```
Comment (MrDHat) [inline utils/event.go:1]: Let's move this to queue package itself

File: queue/queue.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+	streamName string,
+	subject string,
+	jetStreamClient nats.JetStreamContext,
+	nstsClientName string,
+) (Queue, error) {
+	s := &queue{
+		streamName:      streamName,
+		subject:         subject,
+		jetStreamClient: jetStreamClient,
+		nstsClientName:  nstsClientName,
```
Comment (MrDHat) [inline queue/queue.go:181]: nats

**Review by MrDHat — CHANGES_REQUESTED** [review]

**Review by krishnamurthypranesh — APPROVED** [review]

**Review by MrDHat — APPROVED** [review]

File: utils/common.go [hunk 3]
```diff
@@ -42,6 +43,52 @@ func GetTimeDiffInSentence(expiresOn int64) string {
...
+		unit = "minute"
+		qty = minutes
+
+		if minutes != 1 {
+			unit += "s"
+		}
+	}
+
+	if hours >= 1 {
+		unit = "hour"
```
Comment (aashishAil) [inline utils/common.go:72]: we should transfer it to package ```utils/time``` so that it is easily mockable

**Review by aashishAil — APPROVED** [review]

File: utils/common.go (hunk 3 above)
Comment (krishnamurthypranesh) [reply utils/common.go:72]: Done

**Pull request merged**